src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -598,11 +598,8 @@ void ShenandoahConcurrentGC::op_init_mark() {
 
 
   if (heap->mode()->is_generational()) {
-    if (_generation->is_young() || (_generation->is_global() && ShenandoahVerify)) {
-      // The current implementation of swap_remembered_set() copies the write-card-table
-      // to the read-card-table. The remembered sets are also swapped for GLOBAL collections
-      // so that the verifier works with the correct copy of the card table when verifying.
-      // TODO: This path should not really depend on ShenandoahVerify.
+    if (_generation->is_young()) {
```

Comment:
My recollection is that ShenandoahVerify (at least one time) would verify the remembered set even for a global GC.  So that's why we needed to do this, to make the verifier happy.  Has this changed?

Reply:
Your description of the PR says no longer necessary, so I guess that answers my question.

Reply:
I assume the "changes to verification" use the correct copy of the card-table. I'd like us to be sure of that because a clean card table (with no marks) will trivially pass verification.

Reply:
Can you check if there's a similar issue in `ShenandoahDegenGC::op_degenerated()` too?

Reply:
Good catch @ysramakrishna . Updated PR and will wait for tests to complete.